samcli/commands/local/lib/swagger/parser.py

```diff
@@ -305,12 +317,11 @@ def get_routes(self, event_type=Route.API) -> List[Route]:
                 payload_format_version = self._get_payload_format_version(method_config)
 
                 authorizers = method_config.get(SwaggerParser._SWAGGER_SECURITY, None)
+
                 authorizer_name = None
+                use_default_authorizer = True
```

Comment:
Do we enforce having an authorizer by default?

Reply:
Cloudformation behaviour is if there is a default authorizer defined (this is defined at the root swagger level or explicitly if defined a Serverless API resource is being used), then all routes will use that authorizer unless explicitly told not to (either by setting `security` to an empty list in swagger, or setting `NONE` if using Cloudformation/Serverless resources.

So I guess that yes, we are "enforcing" a default authorizer, just by extension of replicating CFN. This would be breaking in the sense that once this feature is released, customers will now find that their Lambda authorizers are being invoked.